packages/core/admin/ee/server/services/user.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const _ = require('lodash');
+
+/** Checks if ee disabled users list needs to be updated
+ * @param {string} id
+ * @param {object} input
+ */
+const shouldUpdateEEDisabledUsersList = async (id, input) => {
+  const data = await strapi.db.query('strapi::ee-store').findOne({
+    where: { key: 'ee_disabled_users' },
+  });
+
+  if (!data || !data.value || data.value.length === 0) return;
+  const disabledUsers = JSON.parse(data.value);
+  const user = disabledUsers.find((user) => user.id === Number(id));
+  if (!user) return;
```

Comment:
We are trying to update a user record, which is held in ee-store. This is not the actual user record, but the one we keep for tracking the enforcement of seats. This condition is here in case ee-store db was altered manually, in which case the update operation would throw.